src/docmanager/widgetmanager.ts

```diff
@@ -69,29 +65,32 @@ class DocumentWidgetManager implements IDisposable {
   }
 
   /**
-   * Test whether the context has been disposed (read-only).
+   * Test whether the document widget manager is disposed.
    */
   get isDisposed(): boolean {
     return this._registry === null;
   }
 
   /**
-   * Dispose of the resources used by the widget manager.
+   * Dispose of the resources used by the widget.
    */
   dispose(): void {
     if (this.isDisposed) {
       return;
     }
+    disconnectReceiver(this);
     this._registry = null;
-    clearSignalData(this);
   }
 
   /**
    * Create a widget for a document and handle its lifecycle.
    */
-  createWidget(name: string, context: DocumentRegistry.IContext<DocumentRegistry.IModel>, kernel?: Kernel.IModel): Widget {
+  createWidget(name: string, context: DocumentRegistry.IContext<DocumentRegistry.IModel>): Widget {
     let factory = this._registry.getWidgetFactory(name);
-    let widget = factory.createNew(context, kernel);
+    if (!factory) {
+      throw new Error(`Factory is not registered: ${name}`);
```

Comment:
Since `throw`s are so rare in the application, maybe the doc string should have a `@throws` description.